### DISCUSSION TITLE
Is it possible to edit the ImageUri value generated by DockerImageFunction?

### DISCUSSION THREAD

**Discussion by WayneLIT** [aws/aws-cdk#29091] [Q&A] [answered]
I am trying to use a container registry located in a different account.

I have set `image_assets_repository_name` and `image_asset_publishing_role_arn` setting appropriately when instantiating my `DefaultStackSynthesizer` and have the appropriate permissions in place.

When performing a `cdk deploy`to account A (with account A credentials), the image is successfully created and pushed to ECR in Account B. However the cloudformation update fails as it cannot find the image. Upon investigation, I noticed in the synthesized template that the `ImageUri` value is prefixed with Account A details:

`"ImageUri": {
      "Fn::Sub": "ACCOUNT-A.dkr.ecr.eu-west-1.${AWS::URLSuffix}/cdk-Qualifier-container-assets-ACCOUNT-B-eu-west-1:f3193b7321a34e0685a8fb9982ce250e2e6d8b45458d55b7c5c42132a7fe60ef"
     }`

So Cloudformation is trying to find a repository and image that does not exist in account A

**Comment by WayneLIT** [comment, marked as answer]
I eventually came to a workaround.

I could find no way to change the account number for the ImageUri field. Looks like it takes it from the account details passed in the `env` dict when instantiating the stack.

The only way around this was to use the lower level CfnFunction resource instead of DockerImageFunction. I created the docker asset separately and constructed the desired imageUri field. In the future this could be included in an internal construct or synthesizer but for now it is a reasonable approach


```
code_asset = DockerImageAsset(self, "MyBuildImage", directory="pipeline/lambda")

image_uri = (
    f"ACCOUNT-B.dkr.ecr.{self.region}.",
    f"{self.url_suffix}/cdk-Qualifier-container-assets-ACCOUNT-B-",
    f"{self.region}:{code_asset.asset_hash}",
)
_lambda.CfnFunction(
    self,
    "DeployLambda",
    code=_lambda.CfnFunction.CodeProperty(image_uri=image_uri),
    timeout=30,
    memory_size=512,
    role=lambda_role.role_arn,
    package_type="Image",
)

```

**Comment by gshpychka** [comment]
Sounds like a bug, consider opening an issue
